datafusion/core/src/physical_optimizer/projection_pushdown.rs

```diff
@@ -454,7 +454,8 @@ fn try_swapping_with_sort(
 
     Ok(Some(Arc::new(
         SortExec::new(updated_exprs, make_with_child(projection, sort.input())?)
```

Comment:
I suggest calling `new_with_children` here https://docs.rs/datafusion/latest/datafusion/physical_plan/trait.ExecutionPlan.html#tymethod.with_new_children as that both:
1. [Does the right thing](https://docs.rs/datafusion-physical-plan/33.0.0/src/datafusion_physical_plan/sorts/sort.rs.html#844-850)
2. Is more likely to be kept up to date other fields are added to SortExec in the future

Reply:
Ah, but I see this API doesn't permit rewriting new exprs 🤔

Reply:
I wonder if we can remove the `preserve_partitioning` parameters because the description of `preserve_partitioning` is not correct now and seem like no sense.
https://github.com/apache/arrow-datafusion/blob/95ba48bd2291dd5c303bdaf88cbb55c79d395930/datafusion/physical-plan/src/sorts/sort.rs#L726-L732

what ever set `preserve_partitioning` true or false, we only sort each partition individually. And the merge work is done in `SortPreservingMergeExec`.

Reply:
Maybe it is acting like a hint to the EnforceDistribution or EnforceSorting passes 🤔 It might be good to investigate further